lib/cylc/job.sh

```diff
@@ -145,7 +145,9 @@ cylc__job__main() {
     # Send task succeeded message
     wait "${CYLC_TASK_MESSAGE_STARTED_PID}" 2>'/dev/null' || true
     cylc message -- "${CYLC_SUITE_NAME}" "${CYLC_TASK_JOB}" 'succeeded' || true
-    trap '' EXIT
+    for signal_name in ${CYLC_VACATION_SIGNALS:-} ${CYLC_FAIL_SIGNALS}; do
+        trap '' "${signal_name}"
+    done
```

Comment:
@dpmatthews discovered that we can actually do:

```bash
trap '' ${CYLC_VACATION_SIGNALS:-} ${CYLC_FAIL_SIGNALS}
```

The syntax appears to work in both `bash` and `ksh`, so good for us. (Not sure why we did what we did before!?)

Reply:
OK, will remove all for; trap loops.